### ISSUE TITLE
Implement Testing

### ISSUE THREAD

**Issue by JoshuaKSmith17** [JoshuaKSmith17/FocalPointDMSClient#12]
Ok, this one is a pretty big deal.  I have never really implemented a testing suite for an application.  This project is probably the best time to start.  This issue is here as a reminder to research the options available and get something going,  Current focus should be on unit testing.

**Comment by JoshuaKSmith17** [comment]
I am researching a solution for this issue now.  I decided to go with NUnit as the testing framework of choice.  It seems like xUnit solutions are the testing framework of choice.  I understand that it is counterintuitive to test classes that have already been created.  If I want to continue with TDD, it makes sense to do this sooner rather than later.

I will start off by creating the testing project and getting a simple test up and running for one of these classes.  I am thinking the best candidate for a first test will be a DataTableBuilder class.  I should also be able to create tests for my factories as well.  

My services classes will require some refactoring if I am to test them properly.  Many of the methods don't return anything, but they could return a Task<> or ValueTask<>.  I initially handled the Task<> within the method body due to issues inferring they EntityType as a return.  I made some tweaks in other areas since then, namely transition from using a marker interface for all database classes to implement to being a subclass of one database object superclass.  Maybe using the superclass will allow type inferencing to work outside the method body as a return.
